features/prog.go

```diff
@@ -151,6 +151,12 @@ func haveProgramType(pt ebpf.ProgramType) error {
 	case errors.Is(err, unix.EPERM):
 		break
 
+	// ENOTSUPP means the program type is at least known to the kernel.
+	case errors.Is(err, unix.ENOTSUPP):
+		if pt == ebpf.StructOps {
+			err = nil
+		}
+
```

Comment:
Should we add a `fallthrough` here in case `ENOTSUPP` happens unexpectedly? Same question for the map type probe. If it happens somewhere unexpectedly we would return it unwrapped.

Reply:
`fallthrough` hurts code readability somewhat, maybe it's probably time to drop the EPERM case and rewrite all outgoing errors in a `defer`, but I expect that approach to be met with resistance as well. :stuck_out_tongue:

Reply:
I've made a separate proposal for this at https://github.com/cilium/ebpf/pull/749. Drafted this one for the time being.